subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdint.h>
+
+#include <toolchain.h>
+#include <sys/util.h>
+
+#include "hal/ccm.h"
+#include "hal/radio.h"
+#include "hal/ticker.h"
+
+#include "util/util.h"
+#include "util/memq.h"
+
+#include "pdu.h"
+
+#include "lll.h"
+#include "lll_vendor.h"
+#include "lll_clock.h"
+#include "lll_chan.h"
+#include "lll_sync.h"
+
+#include "lll_internal.h"
+#include "lll_tim_internal.h"
+#include "lll_prof_internal.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
+#define LOG_MODULE_NAME bt_ctlr_lll_sync
+#include "common/log.h"
+#include <soc.h>
+#include "hal/debug.h"
+
+static int init_reset(void);
+static int prepare_cb(struct lll_prepare_param *p);
+static void isr_rx(void *param);
+
+int lll_sync_init(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int lll_sync_reset(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
```

Comment:
This is a pattern followed so that in the future initialization specific to power and hci reset can be differentiated. The existing code is optimized out by compiler currently.